src/group_theory/torsion.lean

```diff
@@ -75,3 +77,49 @@ exponent_exists_iff_ne_zero.mpr $
 @[to_additive is_add_torsion_of_fintype]
 lemma is_torsion_of_fintype [fintype G] : is_torsion G :=
 exponent_exists.is_torsion $ exponent_exists_iff_ne_zero.mpr exponent_ne_zero_of_fintype
+
+section infinite_torsion_groups
+
+/-- The group ℚ ⧸ ℤ. -/
+@[reducible] def qmodz := ℚ ⧸ (algebra.linear_map ℤ ℚ).range
+
+namespace qmodz
+
+open rat
+
+/-- ℚ ⧸ ℤ is a torsion group -/
+lemma is_torsion : add_monoid.is_torsion qmodz :=
+λ qz, quotient.induction_on' qz $ begin
+  intro q,
+  rw is_of_fin_add_order_iff_nsmul_eq_zero,
+  refine ⟨q.denom, q.pos, quotient.eq'.mpr _⟩,
+  by_cases h : q.denom = 0,
+  { rw h, simp only [nsmul_eq_mul, nat.cast_zero, zero_mul], exact quotient.eq'.mp rfl },
+  { use q.num,
+    apply @mul_right_cancel₀ _ _ _ (q.denom : ℚ)⁻¹ _
+      (by simp only [h, ne.def, inv_eq_zero, not_false_iff, nat.cast_eq_zero]),
+    have z' : ↑(q.num) * (↑(q.denom)) ⁻¹ = (q.num : ℚ) / q.denom := by rw div_eq_mul_inv,
+    field_simp [z', mul_comm, ← _root_.mul_assoc], }
+end
```

Comment:
A much shorter proof:
```suggestion
λ q, begin
  induction q using quotient.induction_on',
  simp_rw [is_of_fin_add_order_iff_nsmul_eq_zero, submodule.quotient.mk'_eq_mk,
    ←submodule.quotient.mk_smul, submodule.quotient.mk_eq_zero, linear_map.mem_range,
    algebra.linear_map_apply, algebra_map_int_eq, ring_hom.eq_int_cast, nsmul_eq_mul, mul_comm],
  exact ⟨q.denom, q.pos, q.num, mul_denom_eq_num.symm⟩,
end
```

`mul_denom_eq_num` does all the heavy lifting here.

Reply:
Thank you. Can you teach me anything more about how you figured this out? How e.g. did you know what to give to `simp_rw`?

Reply:
Sure: what happened is that I found `submodule.quotient.mk_eq_zero`, and then tried to work out how to get the goal to contain `submodule.quotient.mk` instead of `quotient.mk'`, and push the `smul` to where I wanted it.

Everything after that was found by `simp?`, and then reordered by hand to fit in `simp_rw`.

Reply:
Aha. Thank you again!